mdoc/src/main/scala/mdoc/internal/markdown/MarkdownBuilder.scala

```diff
@@ -40,31 +41,40 @@ object MarkdownBuilder {
         val ctor = cls.getDeclaredConstructor()
         ctor.setAccessible(true)
         val doc = ctor.newInstance().asInstanceOf[DocumentBuilder].$doc
-        try {
-          doc.build(instrumentedInput)
-        } catch {
-          case e: DocumentException =>
-            val index = e.sections.length - 1
-            val input = sectionInputs(index).input
-            val pos =
-              if (e.pos.isEmpty) {
-                Position.Range(input, 0, 0)
-              } else {
-                val slice = Position.Range(
-                  input,
-                  e.pos.startLine,
-                  e.pos.startColumn,
-                  e.pos.endLine,
-                  e.pos.endColumn
-                )
-                slice.toUnslicedPosition
-              }
-            reporter.error(pos, e.getCause)
-            Document(instrumentedInput, e.sections)
-          case MdocNonFatal(e) =>
-            reporter.error(e)
-            Document.empty(instrumentedInput)
+        val evaluatedDoc = new AtomicReference[Document]()
+        val thread = new Thread {
```

Comment:
I packed it into a Thread and set the classloader we loaded the DocumentBuilder with. Not sure it's the best way to do it, but seems to work fine.

Reply:
I suggest using the option to hide whitespace changes,

Reply:
This is awesome that this is working. Great jump hunting this down. I get the changes, but maybe a silly question, but what class loader what then being used before this?

Reply:
It was the one we used for classloading mdoc I think, which didn't have the classes from dependencies.